projects/step-core/src/lib/directives/clamp-fade.directive.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  AfterViewInit,
+  Directive,
+  ElementRef,
+  HostBinding,
+  NgZone,
+  OnDestroy,
+  Renderer2,
+  input,
+  effect,
+} from '@angular/core';
+
+const FADE_HEIGHT_PX = 24;
+
+@Directive({
+  selector: '[stepClampFade]',
+  standalone: true,
+})
+export class ClampFadeDirective implements AfterViewInit, OnDestroy {
+  lines = input(5);
+  fadeBackground = input<string | undefined>();
+
+  @HostBinding('attr.data-step-clamp-fade') private attribute = '';
```

Comment:
Avoid the usage of decorators. Instead define host binding in directive's declaration:
```
@Directive({
  selector: '[stepClampFade]',
  host: {
     '[attr.data-step-clamp-fade]': '""'
  }
})
```

https://angular.dev/guide/components/host-elements

Angular team's suggest to always prefer host bindings instead of decorators, because decorators are still supported for backward compatibility only